tsl/src/continuous_aggs/create.c

```diff
@@ -1495,14 +1502,7 @@ cagg_validate_query(const Query *query, const bool finalized, const char *cagg_s
 				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 				 errmsg("cannot create continuous aggregate on hypertable with row security")));
 
-	/*
-	 * We need a GROUP By clause with time_bucket on the partitioning
-	 * column of the hypertable.
-	 */
-	Assert(query->groupClause);
-	caggtimebucket_validate(&bucket_info, query->groupClause, query->targetList);
-
-	/* Nested cagg validations. */
+	/* nested cagg validations */
 	if (is_nested)
```

Comment:
Ok. trying to sort this out a little more, to see if anything could go wrong.

The function `caggtimebucket_validate` will update the `bucket_width` and some other fields in `bucket_info` if it is executed. These are used in the code inside the `if (is_nested)` that follows here. The default value for `bucket_info.bucket_width` is -1, which is an invalid value.

The variable `is_nested` is set to true if and only if `rte->relkind == RELKIND_VIEW` in the code block above, but since we call the code at the end of the if-statement, it seems like the code path `bucket_width` is used below this code also contains a call to the function `caggtimebucket_validate`.

The only situation where `bucket_width` and friends can be unmodified is then when it is not a relation or a view. In this case, the bucket info is returned to the calling function, which is then used in a call to `build_union_query`, but this function does not seem to use this information, so tentatively it seems like the change is safe.

Reply:
One thing that is weird is that the field `hpartcolno` is used inside `build_union_query` but this do not trigger an error even though the code in `caggtimebucket_validate` does it.